I get this:
```
raco setup: --- summary of errors ---
raco setup: error: during building docs for <pkgs>/gui-doc/mrlib/scribblings/mrlib.scrbl
raco setup:   query-rows: attempt to write a readonly database
raco setup:     context...:
raco setup:      /nix/store/yg2nxj1xkcrq3b87xi05172gmnjzpqhz-racket-6.1.1/share/racket/collects/db/private/sqlite3/connection.rkt:387:0: handle-status*
raco setup:      /nix/store/yg2nxj1xkcrq3b87xi05172gmnjzpqhz-racket-6.1.1/share/racket/collects/db/private/sqlite3/connection.rkt:347:8
raco setup:      /nix/store/yg2nxj1xkcrq3b87xi05172gmnjzpqhz-racket-6.1.1/share/racket/collects/db/private/sqlite3/connection.rkt:170:4: step method in connection%
raco setup:      /nix/store/yg2nxj1xkcrq3b87xi05172gmnjzpqhz-racket-6.1.1/share/racket/collects/db/private/sqlite3/connection.rkt:158:8: loop
raco setup:      /nix/store/yg2nxj1xkcrq3b87xi05172gmnjzpqhz-racket-6.1.1/share/racket/collects/db/private/sqlite3/connection.rkt:61:4: query1 method in connection%
raco setup:      /nix/store/yg2nxj1xkcrq3b87xi05172gmnjzpqhz-racket-6.1.1/share/racket/collects/db/private/generic/functions.rkt:165:0: query-rows8
raco setup:      /nix/store/yg2nxj1xkcrq3b87xi05172gmnjzpqhz-racket-6.1.1/share/racket/collects/setup/doc-db.rkt:94:3: temp60
raco setup:      /nix/store/yg2nxj1xkcrq3b87xi05172gmnjzpqhz-racket-6.1.1/share/racket/collects/setup/doc-db.rkt:498:2: loop
raco setup:      /nix/store/yg2nxj1xkcrq3b87xi05172gmnjzpqhz-racket-6.1.1/share/racket/collects/setup/doc-db.rkt:89:0: doc-db-key->path22
raco setup:      /nix/store/yg2nxj1xkcrq3b87xi05172gmnjzpqhz-racket-6.1.1/share/racket/pkgs/racket-index/setup/xref.rkt:120:6: try
raco setup:      /nix/store/yg2nxj1xkcrq3b87xi05172gmnjzpqhz-racket-6.1.1/share/racket/pkgs/racket-index/setup/xref.rkt:116:2
raco setup:      /nix/store/yg2nxj1xkcrq3b87xi05172gmnjzpqhz-racket-6.1.1/share/racket/pkgs/scribble-lib/scribble/xref.rkt:65:19: send-arg73
raco setup:      /nix/store/yg2nxj1xkcrq3b87xi05172gmnjzpqhz-racket-6.1.1/share/racket/pkgs/scribble-lib/scribble/base-render.rkt:1114:28
raco setup:      /nix/store/yg2nxj1xkcrq3b87xi05172gmnjzpqhz-racket-6.1.1/share/racket/pkgs/scribble-lib/scribble/base-render.rkt:1114:28
raco setup:      /nix/store/yg2nxj1xkcrq3b87xi05172gmnjzpqhz-racket-6.1.1/share/racket/pkgs/scribble-lib/scribble/core.rkt:24:0: resolve-get/where
raco setup:      /nix/store/yg2nxj1xkcrq3b87xi05172gmnjzpqhz-racket-6.1.1/share/racket/pkgs/scribble-lib/scribble/core.rkt:73:0: resolve-search...
raco setup:   
raco setup:   
make[1]: *** [install-3m] Error 1
make[1]: Leaving directory `/tmp/nix-build-racket-6.1.1.drv-0/racket-6.1.1/src/build'
make: *** [install] Error 2
builder for ‘/nix/store/40ijlbpzlflmvhcs6jfv47c3b3qypr9z-racket-6.1.1.drv’ failed with exit code 2
error: build of ‘/nix/store/40ijlbpzlflmvhcs6jfv47c3b3qypr9z-racket-6.1.1.drv’ failed
```

--- a/pkgs/development/interpreters/racket/default.nix
+++ b/pkgs/development/interpreters/racket/default.nix
@@ -1,62 +1,68 @@
-{ stdenv, fetchurl, cairo, file, fontconfig, glib, gtk, freefont_ttf
-, libjpeg, libpng, libtool, makeWrapper, openssl, pango, sqlite, which, readline } :
+{ stdenv, fetchurl, makeFontsConf, makeWrapper
+, cairo, coreutils, fontconfig, freefont_ttf
+, glib, gmp, gtk, libffi, libjpeg, libpng
+, libtool, mpfr, openssl, pango, poppler
+, readline, sqlite
+}:
+
+let
+
+  fontsConf = makeFontsConf {
+    fontDirectories = [ freefont_ttf ];
+  };
+
+  libPath = stdenv.lib.makeLibraryPath [
+    cairo
+    fontconfig
+    glib
+    gmp
+    gtk
+    libjpeg
+    libpng
+    mpfr
+    openssl
+    pango
+    poppler
+    readline
+    sqlite
+  ];
+
+in
 
 stdenv.mkDerivation rec {
-  pname = "racket";
+  name = "racket-${version}";
   version = "6.1.1";
-  name = "${pname}-${version}";
 
   src = fetchurl {
     url = "http://mirror.racket-lang.org/installers/${version}/${name}-src.tgz";
     sha256 = "090269522d20e7a5ce85d2251a126745746ebf5e87554c05efe03f3b7173da75";
   };
 
-  # Various Racket executables do runtime searches for these.
-  ffiSharedLibs = "${cairo}/lib:${fontconfig}/lib:${glib}/lib:${gtk}/lib:${libjpeg}/lib:"
-                + "${libpng}/lib:${openssl}/lib:${pango}/lib:${sqlite}/lib:"
-                + "${readline}/lib";
+  FONTCONFIG_FILE = fontsConf;
+  LD_LIBRARY_PATH = libPath;
+  NIX_LDFLAGS = "-lgcc_s";
 
-  buildInputs = [ file fontconfig freefont_ttf libtool makeWrapper sqlite which ];
+  buildInputs = [ fontconfig libffi libtool makeWrapper sqlite ];
 
   preConfigure = ''
-    export LD_LIBRARY_PATH=${ffiSharedLibs}:$LD_LIBRARY_PATH
-
-    # Chroot builds do not have access to /etc/fonts/fonts.conf,
-    # but the Racket bootstrap needs a working fontconfig,
-    # so here a simple temporary stand-in is used.
-    mkdir chroot-fontconfig
-    cat ${fontconfig}/etc/fonts/fonts.conf > chroot-fontconfig/fonts.conf
-    sed -e 's@</fontconfig>@@' -i chroot-fontconfig/fonts.conf
-    echo "<dir>${freefont_ttf}</dir>" >> chroot-fontconfig/fonts.conf
-    echo "</fontconfig>" >> chroot-fontconfig/fonts.conf
-
-    # remove extraneous directories from temporary fonts.conf
-    sed -e 's@<dir></dir>@@g' \
-        -e 's@<dir prefix="xdg">fonts</dir>@@g' \
-        -e 's@<dir>~/.fonts</dir>@@g' \
-        -e 's@<cachedir prefix="xdg">fontconfig</cachedir>@@g' \
-        -e 's@<cachedir>~/.fontconfig</cachedir>@@g' \
-        -i chroot-fontconfig/fonts.conf
-
-    export FONTCONFIG_FILE=$(pwd)/chroot-fontconfig/fonts.conf
-
-    cd src
-    sed -e 's@/usr/bin/uname@'"$(which uname)"'@g' -i configure
-    sed -e 's@/usr/bin/file@'"$(which file)"'@g' -i foreign/libffi/configure
+    substituteInPlace src/configure --replace /usr/bin/uname ${coreutils}/bin/uname
+    mkdir src/build
+    cd src/build
   '';
 
   configureFlags = [ "--enable-shared" "--enable-lt=${libtool}/bin/libtool" ];
 
-  NIX_LDFLAGS = "-lgcc_s";
-  enableParallelBuilding = true;
+  configureScript = "../configure";
+
+  enableParallelBuilding = false;
 
   postInstall = ''
     for p in $(ls $out/bin/) ; do
-      wrapProgram $out/bin/$p --prefix LD_LIBRARY_PATH ":" "${ffiSharedLibs}";
+      wrapProgram $out/bin/$p --set LD_LIBRARY_PATH "${LD_LIBRARY_PATH}";
     done
   '';
 
-  meta = {
+  meta = with stdenv.lib; {
     description = "A programmable programming language";
     longDescription = ''
       Racket is a full-spectrum programming language. It goes beyond
@@ -67,10 +73,9 @@ stdenv.mkDerivation rec {
       libraries support applications from web servers and databases to
       GUIs and charts.
     '';
-
     homepage = http://racket-lang.org/;
-    license = stdenv.lib.licenses.lgpl3;
-    maintainers = with stdenv.lib.maintainers; [ kkallio henrytill ];
-    platforms = stdenv.lib.platforms.linux;
+    license = licenses.lgpl3;
+    maintainers = with maintainers; [ kkallio henrytill ];
+    platforms = platforms.linux;
   };
 }

